### DISCUSSION TITLE
How to add `pip` in the venv for Python 3.12

### DISCUSSION THREAD

**Discussion by bebound** [Homebrew/discussions#5600] [Tap maintenance and Homebrew development] [answered]
### Output of `brew config`

```shell
No
```

### Output of `brew doctor`

```shell
No
```

### Description of issue

After https://github.com/Homebrew/brew/pull/16306, `without_pip=false` is removed. Python 3.12 does not install pip in virtual env.
Our formula needs pip. What's the recommended way to add pip into venv?

**Comment by branchv** [comment, marked as answer]
If `pip` is needed at runtime (note of course this is very rare), then it can and should be declared as a normal dependency, so in `pyproject.toml`'s `project.dependencies` or `setup.py`'s `install_requires`

**Comment by bebound** [comment]
Brew calls `pip` with `--no-deps`, I have to manually add `pip` in resource.

https://github.com/Homebrew/brew/blame/cd651094878e88cb710bc24e253bff8b3595258e/Library/Homebrew/formula.rb#L1855

**Reply by branchv** [reply]
Python formulae are expected to use `brew update-python-resources` to generate their `resource` blocks, see https://docs.brew.sh/Formula-Cookbook#python-dependencies
